File: dice-game/advanced_dice.rb
# YOUR ADVANCED CODE, HERE
def DiceGame()
  puts "How many sides does your dice have? "
  sides = gets.chomp

  puts "How many times would you like to roll your pair of dice?"
  num = gets.chomp
  puts ""

  while true
    counter = 0

    while counter < num.to_i
      num1 = rand(sides.to_i) + 1
      num2 = rand(sides.to_i) + 1

      puts "You rolled a #{num1} and a #{num2}. Total: #{num1+num2}"
      counter += 1
    end

    puts "\nRoll Again? (Y/N)> "
    cont_game = gets.chomp
    if cont_game == "N"
      break
    end
    
  end
end

DiceGame()

File: dice-game/advanced_dice.py
def DiceGame():
    import random

    user_input1 = int(raw_input("How many sides does your dice have? "))
    user_input2 = int(raw_input("How many times would you like to roll your pair of dice? "))
    print ""

    while True:
        total = 0
        while total < user_input2:
            num1 = random.randrange(1,user_input1+1)
            num2 = random.randrange(1,user_input1+1)

            print "You rolled a {} and a {}. Total: {}".format(num1,num2,num1+num2)
            total += 1

        cont_game = raw_input("\nRoll Again? (Y/N)> ")
        print ""

        if cont_game == "N":
            break

DiceGame()
